Deprecated setting `dtype` for latest NumPy dev

See [this](https://github.com/mne-tools/mne-python/actions/runs/24248318019/job/70800795772?pr=13833) for example
```
  File "/home/runner/work/mne-python/mne-python/mne/surface.py", line 813, in _call_pyvista
    out = meth(self.pdata, check_surface=False)
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/filters/data_set.py", line 3090, in select_interior_points
    out['selected_points'] = bools
    ~~~^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/dataset.py", line 1531, in __setitem__
    self.point_data[name] = scalars
    ~~~~~~~~~~~~~~~^^^^^^
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/datasetattributes.py", line 261, in __setitem__
    self.set_array(value, name=key)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/_deprecate_positional_args.py", line 243, in inner_f
    return f(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/datasetattributes.py", line 552, in set_array
    vtk_arr = self._prepare_array(data=data, name=name, deep_copy=deep_copy)
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/datasetattributes.py", line 769, in _prepare_array
    data = data.view(np.uint8)
  File "/opt/hostedtoolcache/Python/3.14.3/x64/lib/python3.14/site-packages/pyvista/core/utilities/misc.py", line 369, in __setattr__
    object.__setattr__(self, key, value)
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
DeprecationWarning: Setting the dtype on a NumPy array has been deprecated in NumPy 2.4.
Instead of changing the dtype on an array x, create a new array with x.view(new_dtype)
_________________ test_no_sparse_head[testing_data-pyvistaqt] __________________
mne/gui/tests/test_coreg.py:377: in test_no_sparse_head
    with pytest.warns(RuntimeWarning, match="No low-resolution head found"):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   DeprecationWarning: Setting the dtype on a NumPy array has been deprecated in NumPy 2.4.
E   Instead of changing the dtype on an array x, create a new array with x.view(new_dtype)
```
Seen with a CI that uses (effectively)
```
pip install --upgrade numpy --index-url "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple"
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -269,7 +269,6 @@ filterwarnings = [
   'ignore:.*numpy.ufunc size changed.*:RuntimeWarning', # bogus numpy ABI warning (see numpy/#432)
   'ignore:Call to deprecated method GetData:DeprecationWarning', # emitted by trame-vtk
   'ignore:Passing .{1}N.{1} to ListedColormap is deprecated since:DeprecationWarning', # https://github.com/matplotlib/cmocean/pull/114
-  'ignore:Setting the dtype on a NumPy array has been deprecated in NumPy 2\.\d+\.', # https://github.com/pyvista/pyvista/issues/8484 - triggered by pyvista_ndarray.view() and VTK's C++ code
   'ignore:Setting the shape on a NumPy array has been deprecated in NumPy 2.5.', # https://numpy.org/devdocs/release/2.5.0-notes.html#setting-the-shape-attribute-is-deprecated
   'ignore:The .*interactive_bk.* attribute was deprecated in Matplotlib 3\.9.*:matplotlib.MatplotlibDeprecationWarning', # https://github.com/microsoft/debugpy/issues/1623
   'ignore:pyvista test \w+ image dir.* does not yet exist.  Creating dir.:UserWarning',

